pipelines.py

```diff
@@ -44,12 +41,75 @@ def unet(config, train_mode):
-    unet_weighted.get_step("unet").transformer = PyTorchUNetWeightedStream(
-        **config.unet) if config.execution.stream_mode else PyTorchUNetWeighted(
-        **config.unet)
+    unet_weighted.get_step("unet").transformer = PyTorchUNetWeightedStream(**config.unet) \
+        if config.execution.stream_mode else PyTorchUNetWeighted(**config.unet)
     return unet_weighted
 
 
+def unet_padded(config, train_mode):
+    if train_mode:
+        save_output = False
+        load_saved_output = False
+    else:
+        save_output = False
+        load_saved_output = False
+
+    if train_mode:
+        loader = _preprocessing_single_generator(config, train_mode, False)
+    else:
+        loader = _preprocessing_single_padded_generator(config)
+
+    unet = Step(name='unet',
+                transformer=PyTorchUNetStream(**config.unet) if config.execution.stream_mode else PyTorchUNet(
+                    **config.unet),
+                input_steps=[loader],
+                cache_dirpath=config.env.cache_dirpath,
+                save_output=save_output, load_saved_output=load_saved_output)
+
+    if train_mode:
+        mask_postprocessed = mask_postprocessing(loader, unet, config, save_output=save_output)
+    else:
+        prediction_crop = Step(name='prediction_crop',
+                               transformer=post.PredictionCropStream(
+                                   **config.postprocessor.prediction_crop) if config.execution.stream_mode \
+                                   else post.PredictionCrop(**config.postprocessor.prediction_crop),
+                               input_steps=[unet],
+                               adapter={'images': ([(unet.name, 'multichannel_map_prediction')]), },
+                               cache_dirpath=config.env.cache_dirpath,
+                               save_output=save_output)
+
+        prediction_renamed = Step(name='prediction_renamed',
+                                  transformer=Dummy(),
+                                  input_steps=[prediction_crop],
+                                  adapter={
+                                      'multichannel_map_prediction': ([(prediction_crop.name, 'cropped_images')]), },
+                                  cache_dirpath=config.env.cache_dirpath,
+                                  save_output=save_output)
+        mask_postprocessed = mask_postprocessing(loader, prediction_renamed, config, save_output=save_output)
+
+    output = Step(name='output',
+                  transformer=Dummy(),
+                  input_steps=[mask_postprocessed],
+                  adapter={'y_pred': ([(mask_postprocessed.name, 'images')]),
+                           'y_scores': ([(mask_postprocessed.name, 'scores')])
+                           },
+                  cache_dirpath=config.env.cache_dirpath,
+                  save_output=save_output,
+                  load_saved_output=False)
+    return output
```

Comment:
@jakubczakon I don't think it's DRY code. What do you think about sth like:
`def unet_padded(config, train_mode):
if train_mode:
  return = unet(config, train_mode)
else:
  unet = unet(config, train_mode).get_step("unet")
  unet.get_step("loader").transformer = loaders.ImageSegmentationLoaderInferencePadding(**config.loader_inference_padding)
prediction_crop = ...`
etc

Do you like it? I think it is cleaner, because you can actually see, that you are only changing inference pipeline.

Reply:
@apyskir I love the idea let me refactor